opentelemetry-sdk/tests/metrics/exponential_histogram/test_exponent_mapping.py

```diff
@@ -0,0 +1,378 @@
+    MAX_NORMAL_VALUE,
+    MIN_NORMAL_EXPONENT,
+    MIN_NORMAL_VALUE,
+)
+
+if version_info >= (3, 9):
+    from math import nextafter
+
+
+def rounded_boundary(scale: int, index: int) -> float:
+    result = 2**index
+
+    for _ in range(scale, 0):
+        result = result * result
+
+    return result
+
+
+class TestExponentMapping(TestCase):
+    def test_singleton(self):
+
+        self.assertIs(ExponentMapping(-3), ExponentMapping(-3))
+        self.assertIsNot(ExponentMapping(-3), ExponentMapping(-5))
+
+    def test_exponent_mapping_0(self):
+
+        try:
+            ExponentMapping(0)
+
+        except Exception as error:
+            self.fail(f"Unexpected exception raised: {error}")
+
+    def test_exponent_mapping_zero(self):
+
+        exponent_mapping = ExponentMapping(0)
+
+        # This is the equivalent to 1.1 in hexadecimal
+        hex_1_1 = 1 + (1 / 16)
+
+        # Testing with values near +inf
+        self.assertEqual(
+            exponent_mapping.map_to_index(MAX_NORMAL_VALUE),
+            MAX_NORMAL_EXPONENT,
+        )
+        self.assertEqual(exponent_mapping.map_to_index(MAX_NORMAL_VALUE), 1023)
+        self.assertEqual(exponent_mapping.map_to_index(2**1023), 1022)
+        self.assertEqual(exponent_mapping.map_to_index(2**1022), 1021)
+        self.assertEqual(
+            exponent_mapping.map_to_index(hex_1_1 * (2**1023)), 1023
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(hex_1_1 * (2**1022)), 1022
+        )
+
+        # Testing with values near 1
+        self.assertEqual(exponent_mapping.map_to_index(4), 1)
+        self.assertEqual(exponent_mapping.map_to_index(3), 1)
+        self.assertEqual(exponent_mapping.map_to_index(2), 0)
+        self.assertEqual(exponent_mapping.map_to_index(1), -1)
+        self.assertEqual(exponent_mapping.map_to_index(0.75), -1)
+        self.assertEqual(exponent_mapping.map_to_index(0.51), -1)
+        self.assertEqual(exponent_mapping.map_to_index(0.5), -2)
+        self.assertEqual(exponent_mapping.map_to_index(0.26), -2)
+        self.assertEqual(exponent_mapping.map_to_index(0.25), -3)
+        self.assertEqual(exponent_mapping.map_to_index(0.126), -3)
+        self.assertEqual(exponent_mapping.map_to_index(0.125), -4)
+
+        # Testing with values near 0
+        self.assertEqual(exponent_mapping.map_to_index(2**-1022), -1023)
+        self.assertEqual(
+            exponent_mapping.map_to_index(hex_1_1 * (2**-1022)), -1022
+        )
+        self.assertEqual(exponent_mapping.map_to_index(2**-1021), -1022)
+        self.assertEqual(
+            exponent_mapping.map_to_index(hex_1_1 * (2**-1021)), -1021
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(2**-1022), MIN_NORMAL_EXPONENT - 1
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(2**-1021), MIN_NORMAL_EXPONENT
+        )
+        # The smallest subnormal value in Python is 2 ** -1074 = 5e-324.
```

Comment:
Out of curiosity, why would it be different for python vs other languages if its using IEE754 under the hood?

Reply:
Right, fixed.